src/pages/ReportDetailsPage.js

```diff
@@ -105,7 +105,10 @@ function ReportDetailsPage(props) {
             return items;
         }
 
-        if ((!isUserCreatedPolicyRoom && participants.length) || (isUserCreatedPolicyRoom && (!ReportUtils.isPublicRoom(props.report) || isPolicyMember))) {
+        // The Members page is only shown when:
+        // - The report is not a user created room with participants to show i.e. DM, Group Chat, etc
+        // - The report is a user created room and the room and the current user is a workspace member i.e. non-workspace members should not see this option.
+        if ((!isUserCreatedPolicyRoom && participants.length) || (isUserCreatedPolicyRoom && isPolicyMember)) {
```

Comment:
NAB but should we create a helper function that better describes these checks?

Reply:
With the extra documentation here now it should be easier to understand the intended logic. In this case, it's nice to have the explanation on the report "details" page vs. abstracted somewhere.